File: objectivec-spritekit/Endless Runner/Endless Runner/model/WallSlice.m
//
//  SliceType.m
//  Endless Runner
//
//  Created by Christopher Caleb on 16/04/2014.
//  Copyright (c) 2014 Christopher Caleb. All rights reserved.
//

#import "WallSlice.h"

@implementation WallSlice

- (id)initWithType:(SliceType)type andY:(CGFloat)y;
{
    self = [super init];
    if (self) {
        _type = type;
        _y = y;
        _sprite = nil;
    }
    
    return self;
}

@end

File: swift-spritekit/Endless-Runner/WallSlice.swift
//
//  WallSlice.swift
//  Endless-Runner
//
//  Created by Christopher Caleb on 12/06/2014.
//  Copyright (c) 2014 Christopher Caleb. All rights reserved.
//

import Foundation
import SpriteKit

let WALL_SLICE_WIDTH: CGFloat = 128

enum SliceType: Int {
    case Front = 0, Back, Step, Decoration, Window, Gap
}

class WallSlice {
    
    let type:   SliceType
    let y:      CGFloat
    var sprite: SKSpriteNode?
    
    init(type: SliceType, y: CGFloat) {
        self.type = type
        self.y    = y
    }
}